instructor/common.py

```diff
@@ -0,0 +1,604 @@
+    assignment = models.Assignment.objects.get(pk=assignment_id)
+    CourseForm = modelform_factory(models.Course, fields=['name', 'code'])
+
+    if request.method == 'POST':
+        if 'back' in request.POST:
+            return redirect('common_assignment_modify')
+
+        if 'course_pk' in request.POST:
+            '''change the course for this assignment'''
+            new_course = models.Course.objects.get(pk=request.POST['course_pk'])
+            assignment.course = new_course
+            assignment.save()
+        else:
+            '''change this course's name or code'''
+            form = CourseForm(request.POST, instance=assignment.course)
+            form.save()
+
+        if 'continue' in request.POST:
+            return redirect("common_assignments_edit_strains")
+
+    all_courses = models.Course.objects.all()
+
+    form = CourseForm(instance=assignment.course)
+    return render_to_response('instructor/course_setup.html',
+                              {'courses': all_courses,
+                               'form': form,
+                               'new':  request.session['new']},
+                              context_instance=RequestContext(request))
+
+
+
+def assignments_edit_meta(request):
+    assignment_id = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=assignment_id)
+    AssignmentForm = modelform_factory(models.Assignment, fields=['has_concentration', 'has_temperature',
+                                                                  'has_start_time', 'has_duration',
+                                                                  'has_collection_time'])
+    message = ''
+    if request.method == "POST":
+        form = AssignmentForm(request.POST, instance=assignment)
+        if ( form.is_valid()):
+            message = "Thank you"
+            form.save()
+        else:
+            message = "Something went wrong"
+    else:
+        form = AssignmentForm(instance=assignment)
+    return render_to_response('instructor/assignment_meta.html',
+                              {'form': form,
+                               'message': message,
+                               'assignment': assignment
+                              },
+                              context_instance=RequestContext(request))
+
+
+def assignments_edit_text(request):
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+    message = ''
+    StrainsFormSet = modelformset_factory(models.AssignmentText, extra=1, fields=['title', 'text'], can_delete=True)
+    if request.method == "POST":
+        formset = StrainsFormSet(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.assignment = assignment
+                form.save()
+        else:
+            message = "Something went wrong"
```

Comment:
I haven't worked on this view yet, not sure if we need it at all. I will leave it as it is.